docs/how-to-guides/upgrade-bootstrap-kubelet.md

```diff
@@ -0,0 +1,113 @@
+# Upgrading bootstrap kubelet
+
+## Contents
+
+- [Introduction](#introduction)
+- [Steps](#steps)
+  - [Step 1: Drain the node](#step-1-drain-the-node)
+  - [Step 2: Find out the IP and SSH](#step-2-find-out-the-ip-and-ssh)
+  - [Step 3: Upgrade kubelet on node](#step-3-upgrade-kubelet-on-node)
+  - [Step 4: Verify](#step-4-verify)
+- [Caveats](#caveats)
+
+## Introduction
+
+[Kubelet](https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/) is a daemon
+that runs on every node and it is responsible for managing Pods on the node.
+
+Lokomotive cluster runs two different sets of kubelet processes. Initially, **bootstrap** kubelet
+configured on the node as a `rkt` pod joins the cluster, and then kubelet pod managed using
+DaemonSet (self-hosted kubelet) takes over the bootstrap kubelet. Self-hosted kubelet allows
+seamless updates between Kubernetes patch version for nodes and also to tune nodes configuration
+using tools like `kubectl`.
+
+At the time of writing `lokoctl` cannot update bootstrap kubelet, so this document explains how to
+perform this operation manually.
+
+## Steps
+
+Perform the following steps on each node, one node at a time.
+
+### Step 1: Drain the node
+
+> **Caution:** If you are using a local directory as a storage for a workload, it will be disturbed
+> by this operation. To avoid this move the workload to another node and let the application
+> replicate the data. If the application does not support data replication across instances, then
+> expect downtime.
+
+```bash
+kubectl drain --ignore-daemonsets <node name>
+```
+
+### Step 2: Find out the IP and SSH
+
+Find the IP of the node by visiting the cloud provider dashboard. Then, connect to selected machine
+using SSH.
+
+```bash
+ssh core@<IP Address>
+```
+
+### Step 3: Upgrade kubelet on node
+
+Run the following commands:
+
+> **NOTE**: Before proceeding to other commands, set the `latest_kube` variable to the latest
+> Kubernetes version. Latest Kubernetes version can be found by running this command after a cluster
+> upgrade: `kubectl version -ojson | jq -r '.serverVersion.gitVersion'`.
+
+```bash
+export latest_kube=<latest kubernetes version e.g. v1.18.0>
+sudo sed -i "s|$(grep -i kubelet_image_tag /etc/kubernetes/kubelet.env)|KUBELET_IMAGE_TAG=${latest_kube}|g" /etc/kubernetes/kubelet.env
+sudo systemctl restart kubelet
+sudo journalctl -fu kubelet
+```
+
+Check the logs carefully. If kubelet fails to restart and instructs to do something (e.g. deleting
+some file), follow the instructions and reboot the node:
+
+```bash
+sudo reboot
+```
+
+### Step 4: Verify
+
+**When `disable_self_hosted_kubelet` is `true`**:
+
+Once the node reboots and kubelet rejoins the cluster, output of following command will show new
+version across the node name:
```

Comment:
`version across the node name`.

I didn't understand this

Reply:
Let's say you are done with worker-0 then you will see the updated version across its name:

```console
$ kubectl get nodes
NAME             STATUS    ROLES    AGE     VERSION
suraj-worker-0   Ready     <none>   1d      v1.18.8
suraj-worker-1   Ready     <none>   1d      v1.18.3
suraj-worker-2   Ready     <none>   1d      v1.18.3
```